packages/xo-web/src/xo-app/vm/tab-advanced.js

```diff
@@ -482,11 +489,33 @@ export default class TabAdvanced extends Component {
     getVmsHaValues().then(vmsHaValues => this.setState({ vmsHaValues }))
   }
 
+  state = {
+    cpuMask: map(this.props.vm.cpuMask, number => ({
+      value: number,
+      label: `Core ${number} `,
```

Comment:
This should always come directly from the props via a selector. Thus selector should use `_getCpuMaskOptions` and find the correct options to be added to the list.